### ISSUE TITLE
Modal allows background text selection & scroll

### ISSUE THREAD

**Issue by marchellodev** [lxsmnsyc/solid-headless#19]
When the modal window is open, the user can still scroll the background content (via the mouse wheel) or select all text (via Ctrl+A).

To minimally reproduce this:
1. open https://solid-headless.vercel.app/preview/dialog
2. open dev tools and add a lot of text after the button html code
3. open the modal window
4. try scrolling or pressing Ctrl+A to select all the text

The issue might be related to https://github.com/tailwindlabs/headlessui/issues/1056
Also maybe it has something to do with the fact that I am using astro js, but I do not think so.

**Comment by marchellodev** [comment]
I believe this is due to the fact that `Dialog.Backdrop` (https://headlessui.com/react/dialog) is not implemented

**Comment by marchellodev** [comment]
A temporary fix inspired by https://github.com/tailwindlabs/headlessui/blob/main/packages/%40headlessui-react/src/components/dialog/dialog.tsx (and a lot of googling):

```
  function closeModal() {
    setIsOpen(false);
    document.documentElement.style.overflow = "";
    document.documentElement.style.paddingRight = "";
  }

  function openModal() {
    const scrollbarWidth = window.innerWidth - document.body.offsetWidth
    setIsOpen(true);
    document.documentElement.style.overflow = "hidden";
    document.documentElement.style.paddingRight = scrollbarWidth+'px';
  }
```

**Comment by marchellodev** [comment]
The text is still selectable with Ctrl+A though :(

**Comment by lxsmnsyc** [comment]
Hmmm interesting behavior, I'll take a look

**Comment by marchellodev** [comment]
I have tried adding `pointer-events: none;`, but I can still scroll the background and now when I click on the background, the modal does not close :)

```
<DialogOverlay class="fixed inset-0 bg-white bg-opacity-40 pointer-events-none" />
```

**Comment by marchellodev** [comment]
btw, when the modal is wrapped inside `<dialog>`, Ctrl+A works as it is supposed to:)

**Comment by lxsmnsyc** [comment]
> the modal does not close :)
Right, that went over my head 🤣 so it's not the solution then. Let's see

> Ctrl+A works as it is supposed to:)
Well yes but that's an HTML specification, not an ARIA specification actually. [There's no mention for managing text selection](https://www.w3.org/WAI/ARIA/apg/patterns/dialogmodal/)
